llama_index/embeddings/openai.py

```diff
@@ -249,6 +250,7 @@ def _get_text_embeddings(self, texts: List[str]) -> List[List[float]]:
         Can be overriden for batch queries.
 
         """
+        event_id = self._callback_manager.on_event_start(CBEventType.EMBEDDING)
```

Comment:
nit: why don't we hook this in the base class i.e. in `BaseEmbedding`

Reply:
Openai does actual batch calls for embeddings, but the base class does not.

So a slightly different hook for counting the nodes properly

Reply:
I see, so the semantic of an "event" is a single API request?

Reply:
Yea exactly 👍